### ISSUE TITLE
Add Private API support

### ISSUE THREAD

**Issue by joshporter1** [arxopia/uirusu#11]
I use the [Private API](https://www.virustotal.com/en/documentation/private-api) to get additional metadata from VirusTotal, and it would be nice for this gem to support that. It's just a matter of adding some additional optional parameters and some more API calls. 

I currently have a [fork](https://github.com/joshporter1/uirusu) where I'm working on adding some of these features. 

Here's a list of the additional endpoints and ones that need modification (add optional parameters)
### Modify current calls
- POST /vtapi/v2/file/scan
- POST /vtapi/v2/file/rescan
- GET /vtapi/v2/file/report
- POST /vtapi/v2/url/scan
- GET /vtapi/v2/url/report
- POST /vtapi/v2/comments/put
### Additional calls
- GET /vtapi/v2/file/scan/upload_url
- POST /vtapi/v2/file/rescan/delete
- GET /vtapi/v2/file/behaviour
- GET /vtapi/v2/file/network-traffic
- GET /vtapi/v2/file/search
- GET /vtapi/v2/file/clusters
- GET /vtapi/v2/file/feed
- GET /vtapi/v2/file/download
- GET /vtapi/v2/file/false-positives
- GET /vtapi/v2/url/feed
- GET /vtapi/v2/ip-address/report
- GET /vtapi/v2/domain/report
- GET /vtapi/v2/comments/get

I'm not sure if Private API support is something you'd like to add to your gem, but adding it won't hurt the other functionality and would add to the use cases your gem currently provides.

**Comment by hammackj** [comment]
I'd love to support it. I do not have private API access at the moment. If you want to submit a pull request I can merge it in to a release.

**Comment by joshporter1** [comment]
Great, I'll work on adding the features and testing them and submit a pull request once it's done!

**Comment by joshporter1** [comment]
I've updated the gem to include support for all of the additional private API endpoints. I've also modified the existing functions (mostly the `*/report` endpoints) to accept additional optional parameters in the form of named function parameters (`**args`). This adds support for all optional parameters without having to add nil checks for all params. 

Since I added lots of new methods, I moved the `response.code` case statement into a function named `parse_response`. I also moved the RestClient call to a separate function named `query_api` that performs the check for nil API keys and adds an exception handler around the RestClient call so it actually raises the proper exceptions in `parse_response`. 

Brief summary:
* Added private API functions to VTComment, VTUrl and VTFile
* Modified calls to accept optional params in VTUrl and VTFile
* Added modules VTDomain and VTIPAddr
* Added `parse_response` and `query_api` to Uirusu module

I still need to write some tests for the new code, and modified the existing tests (Actual results for scans are 1 higher than expected now). 

Check out the code [here](https://github.com/joshporter1/uirusu) and let me know if you want me to make any changes. Once I write some tests and get your approval here I'll submit a pull request.
